substrate/frame/revive/src/precompiles/builtin/create2.rs

```diff
@@ -0,0 +1,134 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+	address::AddressMapper,
+	precompiles::{BuiltinAddressMatcher, Error, ExtWithInfo, PrimitivePrecompile},
+	BalanceOf, Config, H256,
+};
+use alloc::{vec, vec::Vec};
+use core::{marker::PhantomData, num::NonZero};
+use sp_core::U256;
+use sp_runtime::DispatchError;
+
+pub struct Create2<T>(PhantomData<T>);
+
+impl<T: Config> PrimitivePrecompile for Create2<T> {
```

Comment:
Wrong trait. `PrimitivePrecompile` is reserved for those: https://www.evm.codes/precompiled

```suggestion
impl<T: Config> BuiltinPrecompile for Create2<T> {
```